models/user/user.go

```diff
@@ -860,15 +861,34 @@ func updateUser(ctx context.Context, u *User, changePrimaryEmail bool) error {
 		}); err != nil {
 			return err
 		}
+	} else { // check if primary email in email_address table
+		primaryEmailExist, err := e.Where("uid=? AND is_primary=?", u.ID, true).Exist(&EmailAddress{})
+		if err != nil {
+			return err
+		}
+
+		if !primaryEmailExist {
+			_, err = e.Insert(&EmailAddress{
+				Email:       u.Email,
+				UID:         u.ID,
+				IsActivated: true,
+				IsPrimary:   true,
+			})
+			return err
```

Comment:
I think this `return err` is unnecessary, and leads to strange results:
If you call updateUser for a User whose Email didn't exist, the method will create the email in the db, not return an error,  and not update the user in the end. Is that really intended?

Reply:
What I would agree with would be
```go
if err != nil {
  return err
}
```